### PR TITLE
Add ICorDebugHeapValue4 -- CreatePinnedHandle

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

File: src/coreclr/src/inc/cordebug.idl [hunk 1]
```diff
@@ -6472,6 +6475,27 @@ interface ICorDebugHeapValue3 : IUnknown
     HRESULT GetMonitorEventWaitList([out] ICorDebugThreadEnum **ppThreadEnum);
 };
 
+/*
```
Comment (gregg-miskelly) [inline src/coreclr/src/inc/cordebug.idl:6478]: Should we also update the documentation for EnableGCNotification to indicate that it is deprecated?

Comment (sdmaclea) [reply src/coreclr/src/inc/cordebug.idl:6478]: Thanks

**Review by gregg-miskelly — APPROVED** [review]
Otherwise LGTM

**Review by chuckries — APPROVED** [review]

**Comment by kouvel** [comment]
Thanks @sdmaclea! LGTM, though I'm not too familiar with adding APIs in this area.

**Review by hoyosjs — APPROVED** [review]

**Comment by sdmaclea** [comment]
@kouvel I expect when you have the rest of the tiered JIT, data breakpoint deadlock fix finished.  You will backport the whole fix to 5.0 servicing branch.

**Comment by kouvel** [comment]
Yep about to put up a PR for my side shortly, will port both once that's in

**Pull request merged**